### PR TITLE
Update PR Template

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Simplifies the PR template and removes the categorization system, as it adds friction and post-hoc categorization is not particularly difficult.

### CODE REVIEW

**Comment by OwenKephart** [comment]
* **#25039** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25039?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by sryza — COMMENTED** [review]
So simple! Love it

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -4,8 +4,4 @@
 
 ## Changelog
 
-Insert changelog entry or "NOCHANGELOG" here.
-
-- [ ] `NEW` _(added new feature or capability)_
-- [ ] `BUGFIX` _(fixed a bug)_
-- [ ] `DOCS` _(added or updated documentation)_
+Insert changelog entry or delete this section.
```
Comment (alangenfeld) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: ## Changelog
> Insert changelog entry or delete this section.

Comment (alangenfeld) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: ^ I think `>` quoting it to make it visually distinct is nice

**Review by alangenfeld — APPROVED** [review]

**Pull request merged**